common/ECChannel.cpp

```diff
@@ -47,12 +47,12 @@ openssl req -new -x509 -key privkey.pem -out cacert.pem -days 1095
 std::atomic<SSL_CTX *> ECChannel::lpCTX{nullptr};
 
 HRESULT ECChannel::HrSetCtx(ECConfig *lpConfig) {
-	HRESULT hr = MAPI_E_CALL_FAILED;
 	if (lpConfig == NULL) {
 		ec_log_err("ECChannel::HrSetCtx(): invalid parameters");
-		return hr;
+		return MAPI_E_CALL_FAILED;
```

Comment:
Going backwards on 487ec771c?

Reply:
In a small sense - basically, this is just so that the basic parameter check for lpConfig doesn't need to jump over the variable declarations (which it couldn't from this point due to some of the variables not being initialized yet, see the GCC compilation error message when changing that to a goto), but there's no harm in pulling the check below the variable declarations which in turn then allows the jump to the end of the method, because all non-basic variables are in scope at the point of this goto. For all others, the variables are in scope. I'll adjust this part of the patch set appropriately.